get-started/index.md

```diff
@@ -62,7 +65,302 @@ the specifics of how to deploy your Cryostat instance. Continue to [Setup](#setu
+
+```
+-Dcom.sun.management.jmxremote.port=9091
+-Dcom.sun.management.jmxremote.ssl=false
+-Dcom.sun.management.jmxremote.authenticate=false
+```
+
+[comment]: # TODO explain how to configure SSL and auth for JMX, or link to external docs
+
+In a real scenario you should enable both SSL and authentication on your application. You can then [trust the certificate](/guides/#add-a-trusted-certificate)
+and [store the credentials](/guides/#store-jmx-credentials).
+
+Depending on your application or its framework, you may set these flags directly in a `Dockerfile` entrypoint, an environment variable, or similar. This may or
+may not require a container image rebuild, and it will require the container to be restarted. Once this is done the application container will be listening for
+incoming JMX connections on port 9091. Let's assume it can be done by setting an environment variable, so we only need to modify our `Deployment:`
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+...
+spec:
+  ...
+  template:
+    ...
+    spec:
+      containers:
+        - name: sample-app
+          image: docker.io/myorg/myapp:latest
+          env:
+            - name: JAVA_OPTS
+              value: "-Dcom.sun.management.jmxremote.port=9091 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false"
+            ...
+```
+
+Next, we need to configure a Kubernetes `Service` to expose this port for cluster-internal traffic, so that Cryostat can see
+and connect to this application JMX port.
+
+```yaml
+apiVersion: v1
+kind: Service
+...
+spec:
+  ports:
+    - name: "jfr-jmx"
+      port: 9091
+      targetPort: 9091
+...
+```
+
+Cryostat queries the Kubernetes API server and looks for `Service`s with a port either named `jfr-jmx` or with the number `9091`. One or both of these conditions
+must be met or else Cryostat will not automatically detect your application. In this case you may wish to use the [Cryostat Agent](#using-the-cryostat-agent-with-jmx)
+to enable discovery, while keeping communications over JMX rather than HTTP.
+
+#### [Using the Cryostat Agent with JMX](#using-the-cryostat-agent-with-jmx)
+The two prior sections have discussed:
+  - How to use the Cryostat Agent to do application discovery and expose data over HTTP.
+  - How to use Kubernetes `Service` configurations for discovery and JMX to expose data.
+
+There is a third, hybrid approach: **using the Cryostat Agent to do application discovery, and JMX to expose data**. This may be
+useful since the Agent HTTP data model is readonly, whereas JMX is read-write. This means that using JMX to communicate between Cryostat and your applications
+allows for more dynamic flexibility, for example, the ability to start and stop Flight Recordings on demand. Using the Cryostat Agent for application discovery
+is also more flexible than depending on `Services` with specially-named or specially-numbered ports. 
```

Comment:
> than depending on `Services` with

Previously, we use `Service`s (the s is outside the inline code). We should be doing the same here.